### ISSUE TITLE
How to define functions

### ISSUE THREAD

**Issue by chris-martin** [typeclasses/haskell-phrasebook#8]
For inspiration, Go By Example has a page on [defining functions](https://gobyexample.com/functions). Go By Example also has a separate page on [multiple return values](https://gobyexample.com/multiple-return-values), but perhaps we could combine the two into one page.

Here are the beginnings of some multiple-return-values ideas:

```haskell
vals = (3, 7)

f x = (x, x + 1)

main =
  do
    let (a, b) = vals
    putStrLn (show a)
    putStrLn (show b)

    let (_, c) = f 4
    putStrLn (show c)
```

**Comment by shubhamkumar13** [comment]
I had a question, Can we talk about type signatures while describing how defining functions work?

Edit : also how currying works? Because that's something cool about functions in haskell

**Comment by ammatsui** [comment]
How currying works
[funcurry.txt](https://github.com/typeclasses/haskell-phrasebook/files/3676736/funcurry.txt)

**Comment by chris-martin** [comment]
Hey thanks for the PR! Will address your questions on #16

**Comment by argumatronic** [comment]
Closed with PR #16.
